priyanka/week4 - LinkedList/delete-middle-node.py

```diff
@@ -0,0 +1,87 @@
+# Definition for singly-linked list.
+class ListNode:
+    def __init__(self, val=0, next=None):
+        self.val = val
+        self.next = next
+
+## UMPIRE: 
+# U: undestand  :
+#. Practice: ask questions and discuss corner cases:
+# M: Match :
+# practice: always match with known DS or Algorithms to solve problem , run with an example
+#P: Planning / pseudocode : 
+# practice: come up with steps on how to solve/approach problem
+# funtion length of lll
+# d middle node :
+#  length=  fun(len):
+#    mi = l //2
+## I: Implement
+## R: Revise: Current Implementation:
+#  Time complexity O(n) as i need to perform 2 traversals ( finding length and delete moddle node doing another traversal)
+
+## Improvements : May be instead of 2 traversals can reduce to 1 travesal by using double pointers
+# Space Complexity O(1)
+
+class Solution:
+    def lenll(head):
+        #ll None
+        i = 0
+        while head != None:
+            head = head.next
+            i += 1
+        return i
+    
+    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
+        ### Corner cases 1. if node is empty
+        if head == None or head.next == None:
+            return None
+        ## 2. if list has only 1 node
+        # if head.next == None:
+        #     return None
+        
+        current = head
+        #length of LL
+        i = Solution.lenll(head)
+        mid_len = i // 2
+
+        ## delete middle node while looping through list
```

Comment:
This is a `what` comment that doesn't really help understand what the code does. Here is a better version:
```
# Find the node right before the middle one so we can delete it.
```